### PR TITLE
Add SetTemplateIcon support and example PDF

### LINKED ISSUE #45: Use templated icons for the menu bar in macOS
Since Yosemite (10.10+), the menubar supports templated images, so that macOS can automatically change the highlight for dark/light themed menu bars.

menubar-light
menubar-dark

Apple recommends using a Template Icon for the menu bar extras:
> **Use a template image to represent your menu bar extra.** A template image discards color information and uses a mask to produce the appearance you see onscreen. Template images automatically adapt to the user’s appearance settings, so they look good on both dark and light menu bars, and when your menu bar extra is selected.

https://developer.apple.com/macos/human-interface-guidelines/extensions/menu-bar-extras

### PR DESCRIPTION
**Pull request by flowchartsman**
Fixes #45

### CODE REVIEW

File: systray_nonwindows.go [hunk 1]
```diff
@@ -31,6 +31,14 @@ func SetIcon(iconBytes []byte) {
 	C.setIcon(cstr, (C.int)(len(iconBytes)))
 }
 
+// SetTemplateIcon sets a templated systray icon for OSX. It is only available
+// on Mac. iconBytes should ideally be a resolution-independent pdf, and should
+// consist of black and clear colors (and an alpha  channel).
+func SetTemplateIcon(iconBytes []byte) {
+	cstr := (*C.char)(unsafe.Pointer(&iconBytes[0]))
+	C.setTemplateIcon(cstr, (C.int)(len(iconBytes)))
```
Comment (stoggi) [inline systray_nonwindows.go]: What happens on Linux? Does it compile if the implementation of SetTemplateIcon is missing from `systray_linux.c`? Maybe we need to add an implementation there too, that just calls SetIcon?

**Comment by flowchartsman** [comment]
Oh I bet it fails! I was kind of being lazy and hoping someone else would do it, haha. I’ll fix it up later. Ideally, it would be stubbed on all supported platforms so the example could build.

**Comment by stoggi** [comment]
Ah, hehe. Making it cross platform was also why I hadn't gotten around to it.

**Comment by flowchartsman** [comment]
I believe I can restructure the repo such that only functions that are supported on a given platform are exposed on that platform.  Would you like me to do that instead (before I put in the work stubbing this out on different platforms)?

**Comment by myleshorton** [comment]
> I believe I can restructure the repo such that only functions that are supported on a given platform are exposed on that platform.

Sounds promising to me. Just making them a no-op where not supported also seems totally adequate.

**Comment by mk23** [comment]
:+1 on getting this integrated.  Tray icons look terrible (or disappear entirely) on Mojave in dark mode.

**Comment by mk23** [comment]
Another option suggestion:  Check if the image is monochromatic and set it as a template.  This would address several stated concerns.  Here's the proof of concept (tested with various b&w and color images).
```diff
diff --git a/systray_darwin.m b/systray_darwin.m
index 3d92868..a4a46cb 100644
--- a/systray_darwin.m
+++ b/systray_darwin.m
@@ -196,6 +196,9 @@ void setIcon(const char* iconBytes, int length) {
   NSData* buffer = [NSData dataWithBytes: iconBytes length:length];
   NSImage *image = [[NSImage alloc] initWithData:buffer];
   [image setSize:NSMakeSize(16, 16)];
+  if ([image.representations[0].colorSpaceName isEqualToString: NSCalibratedWhiteColorSpace]) {
+    image.template = YES;
+  }
   runInMainThread(@selector(setIcon:), (id)image);
 }
```

**Comment by aleksandrzhiliaev** [comment]
is there anything that I can help with?
this feature is really needed

**Comment by oxtoacart** [comment]
Closing in favor of #119

**Pull request closed without merging**